src/FileFormats/MOF/read.jl

```diff
@@ -25,6 +25,48 @@ function Base.read!(io::IO, model::Model)
     name_map = read_variables(model, object)
     read_objective(model, object, name_map)
     read_constraints(model, object, name_map)
+    _convert_to_nlpblock(model)
+    return
+end
+
+function _convert_to_nlpblock(model::Model)
+    has_constraints = false
+    nlp_model = MOI.Nonlinear.Model()
+    for S in (
+        MOI.LessThan{Float64},
+        MOI.GreaterThan{Float64},
+        MOI.EqualTo{Float64},
+        MOI.Interval{Float64},
+    )
+        for ci in MOI.get(model, MOI.ListOfConstraintIndices{Nonlinear,S}())
+            f = MOI.get(model, MOI.ConstraintFunction(), ci)
+            set = MOI.get(model, MOI.ConstraintSet(), ci)
+            MOI.Nonlinear.add_constraint(nlp_model, f.expr, set)
+            # We don't need this in `model` any more.
+            MOI.delete(model, ci)
+            has_constraints = true
+        end
+    end
+    if MOI.get(model, MOI.ObjectiveFunctionType()) == Nonlinear
+        obj = MOI.get(model, MOI.ObjectiveFunction{Nonlinear}())
+        MOI.Nonlinear.set_objective(nlp_model, obj.expr)
+        MOI.set(
+            model,
+            MOI.ObjectiveFunction{MOI.ScalarAffineFunction{Float64}}(),
+            MOI.ScalarAffineFunction{Float64}(
+                MOI.ScalarAffineTerm{Float64}[],
+                0.0,
+            ),
+        )
+    end
+    if has_constraints
+        evaluator = MOI.Nonlinear.Evaluator(
+            nlp_model,
+            MOI.Nonlinear.SparseReverseMode(),
```

Comment:
Does this mean that only the `SparseReverseMode` AD backend can be used with reading from an MOF file?

Reply:
I guess we can add this as an option

Reply:
Added